includes/Services/StatusService.php

```diff
@@ -29,10 +33,82 @@ public static function handle_started() {
 	public static function handle_completed() {
 		if ( 'started' === get_option( Options::get_option_name( 'status' ) ) ) {
 			update_option( Options::get_option_name( 'status' ), 'completed' );
+			self::update_onboarding_restart_status();
 			do_action( 'newfold/onboarding/completed' );
 		}
 	}
 
+	/**
+	 * Checks if the WordPress site was created within the last 9 months (275 days)
+	 * using the 'bluehost_plugin_install_date' option.
+	 *
+	 * @return bool True if the site was created within the last 275 days, false otherwise.
+	 */
+	private static function is_site_created_within_last_9_months() {
+		$install_date_timestamp = get_option( Options::get_option_name( 'bluehost_plugin_install_date', false ) );
+
+		// If the option is not set or is invalid, return false
+		if ( ! $install_date_timestamp ) {
+			return false;
+		}
+
+		// Calculate the timestamp for 275 days ago (9 months)
+		$nine_months_ago = time() - ( 275 * 24 * 60 * 60 );
+
+		return $install_date_timestamp >= $nine_months_ago;
+	}
+
+	/**
+	 * Checks if the user is eligible to restart onboarding based on brand configuration and AI SiteGen capability.
+	 *
+	 * @return bool True if eligible, false otherwise.
+	 */
+	public static function is_onboarding_restart_eligible() {
+		// Check if the brand is eligible for Restarting Onboarding
+		$brand_config = Brands::get_brands()[ NFD_ONBOARDING_PLUGIN_BRAND ]['config'] ?? array();
+		if ( empty( $brand_config['canRestartOnboarding'] ) || ! $brand_config['canRestartOnboarding'] ) {
+			return false;
+		}
+
+		// Check if AI SiteGen Hiive capability is active and the site was created in the last 9 months
+		if ( ! Config::has_ai_sitegen() || ! self::is_site_created_within_last_9_months() ) {
+			return false;
+		}
+
+		return true;
+	}
+
+	/**
+	 * Handles the flow data and updates the restart eligibility status based on total onboarding tries.
+	 *
+	 * @return void
+	 */
+	public static function update_onboarding_restart_status() {
+
+		// Don't do anything if the customer is not eligible
+		if ( ! self::is_onboarding_restart_eligible() ) {
+			return;
+		}
+
+		// Get flow data
+		$flow_data = get_option( Options::get_option_name( 'flow' ) );
+
+		if ( isset( $flow_data ) ) {
+			// Increment the total onboarding tries
+			$flow_data['onboardingRetries']['retryCount'] = ( $flow_data['onboardingRetries']['retryCount'] ?? 0 ) + 1;
+
+			// Update the flow data with the incremented total onboarding tries count
+			update_option( Options::get_option_name( 'flow' ), $flow_data );
+
+			// Determine eligibility for restarting onboarding
+			$current_retry_count = $flow_data['onboardingRetries']['retryCount'];
+			$can_restart        = $current_retry_count < $flow_data['onboardingRetries']['maxRetryCount'];
```

Comment:
If i'm reading the ticket correctly [8-243](https://jira.newfold.com/browse/PRESS8-243), it states "If Onboarding has been re-tried three or fewer times"

this check only pass if the `$current_retry_count` is less that `$flow_data['onboardingRetries']['maxRetryCount']` '3'... Should we check for equal or smaller than?

Reply:
So basically why I wrote this was, 

**_First_** Go the user is automatically redirected to Onboarding and completes/exists.
Then the count function runs and changes the value to **1**

**_Secondly_**, when the user clicks on Build with AI, The flow starts again and after the flow is completed/exited it changes to **2**

**_Thirdly_**, Again the user can go for onboarding as 2 < 3 `($current_retry_count < maxRetryCount)` and after onboarding is completed/exited it reaches this function and changes the value to **3**. 

Now as the **three tries** are over the value for `($current_retry_count < maxRetryCount)` i.e. `3 < 3` is **_false_** due to which `can_restart` saves false and the user cannot go through onboading again.